diff --git a/src/common/gnomesettings.cpp b/src/common/gnomesettings.cpp
index 95d572f..a028570 100644
--- a/src/common/gnomesettings.cpp
+++ b/src/common/gnomesettings.cpp
@@ -265,7 +265,13 @@ static QString colorScheme(bool useHighContrast, bool useDarkVariant)
         colorScheme = useDarkVariant ? QStringLiteral("AdwaitaDark") : QStringLiteral("Adwaita");
     }
 
-    return QStandardPaths::locate(QStandardPaths::GenericDataLocation, QStringLiteral("color-schemes/") + colorScheme + QStringLiteral(".colors"));
+    QString colorSchemePath = QDir::cleanPath(QCoreApplication::applicationDirPath() + QStringLiteral("/../share/color-schemes/"));
+    QString file = colorScheme + QStringLiteral(".colors");
+    if (QDir(colorSchemePath).exists(file)) {
+        return colorSchemePath + file;
+    } else {
+        return QStringLiteral("");
+    }
 }
 
 void GnomeSettings::loadPalette()

